File: solutions/122. Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 0)
            return 0;
​
        int sum = 0, prev = 0;
        for (int i = 0; i < prices.size(); i++)
        {
            if (i == prices.size() - 1)
            {
                sum += prices[i] - prices[prev];
            }
​
            else if (prices[i] > prices[i + 1])
            {
                sum += prices[i] - prices[prev];
                prev = i + 1;
            }
        }
​
        return sum;
    }
};

File: ..solutions/122. Best Time to Buy and Sell Stock II.java
class Solution {
    public int maxProfit(int[] arr) {
        final int n = arr.length;
        int overrallProfit = 0;
​
        for (int i = 1; i < n; i++) {
            int profit = arr[i] - arr[i-1];
            if (profit > 0) {
                overrallProfit += profit;
            }
        }
        
        return overrallProfit;
    }
}
